### ISSUE TITLE
TransOver isn't working anymore.

### ISSUE THREAD

**Issue by Jerzech** [artemave/translate_onhover#40]
Hi, I really appreciate your work and I love this extension, but a couple days ago it's stopped working.I tried changing the setting, reinstalling and installing older versions but nothing helped.

**Comment by artemave** [comment]
Works for me. I have people occasionally complaining that it just stops working... Could you inspect "the background page" for me?

To do this, right click on TransOver button to see the details and once there click "inspect background page" (or however it is called in your language). Here is the screenshot:
![image](https://user-images.githubusercontent.com/23721/43339225-af4148d4-91d0-11e8-86e5-571d3ce23ca9.png)

You should see a page that looks like this:
![image](https://user-images.githubusercontent.com/23721/43339348-faae9dbc-91d0-11e8-9489-1a9e680863a6.png)

We're looking for any red text among those log lines. If there is, it'll likely tell us more about the problem. Perhaps you could post here a screenshot?

**Comment by Jerzech** [comment]
Now it work from time to time and here are the errors:
![image](https://user-images.githubusercontent.com/24878855/43344614-b3409902-91ea-11e8-80ac-1ac20d3f8c64.png)

**Comment by artemave** [comment]
Wait... Is this Vivaldi browser? According to this Vivaldi had some problems with `detectLanguage` (the line of code that the error points to)

**Comment by Jerzech** [comment]
Yea it's, but i also tested it in chrome and it wasn't working there either.
![image](https://user-images.githubusercontent.com/24878855/43347872-85d41de2-91f7-11e8-89ee-415049b0d6f4.png)

**Comment by artemave** [comment]
Right. Could you click on one of those triangles next to GET? Maybe they'll tell us more. Also, could you switch to the network tab and screenshot the headers of one of those failed requests?

503 status code means "service unavailable" (google translate API that is) which could be because you exceeded some sort of a rate limiter of theirs. We might be able to see it in the response headers.

**Comment by Jerzech** [comment]
I couldn't exceed any limiter because I installed TransOver in Chrome only to check if it is working there, unless this limit is shared between browsers.
![image](https://user-images.githubusercontent.com/24878855/43354728-c11bc0e4-9251-11e8-835e-54f13c69bd17.png)

There are two different errors in the network tab:
![image](https://user-images.githubusercontent.com/24878855/43354754-0d047a0a-9252-11e8-8399-a3603c11f03b.png)

![image](https://user-images.githubusercontent.com/24878855/43354768-5f387cc2-9252-11e8-9eee-f45abbb2b018.png)

**Comment by artemave** [comment]
No clues there sadly. 

The limit would apply to your ip, not the browser. So it wouldn't matter when you install Chrome. On that note, have you got a vpn account? It'd be interesting to test if the error still occurs once you're on vpn (as it changes your internet "location")

**Comment by atrox3d** [comment]
hello, i'm having the same problem.
Clicking the links in the devtools backgroundpage  [https://translate.googleapis.com/translate_a/single?dt=t&dt=bd&client=gtx&q=keyword%20&sl=auto&tl=hu&dj=1&source=bubble#](url) I get this message (in italian):


"I nostri sistemi hanno rilevato un traffico insolito proveniente dalla rete del tuo computer. Prova a inviare nuovamente la tua richiesta più tardi. A che cosa è dovuto?

Questa pagina viene visualizzata quando Google rileva automaticamente delle richieste provenienti dalla rete del tuo computer che apparentemente violano i Termini di servizio. Il blocco scadrà quando tali richieste saranno terminate. 

Questo traffico potrebbe essere stato inviato da software dannoso, da un plug-in del browser o da script che invia richieste automatiche. Se condividi la connessione di rete, rivolgiti all'amministratore per assistenza: la responsabilità potrebbe essere di un altro computer che utilizza lo stesso indirizzo IP. Ulteriori informazioni

A volte questa pagina viene visualizzata se stai impiegando termini avanzati che vengono solitamente utilizzati dai robot o se stai inviando richieste molto rapidamente.

Indirizzo IP: ***.***.***.***
Data: 2018-07-29T18:19:37Z
URL: https://translate.googleapis.com/translate_a/single?dt=t&dt=bd&client=gtx&q=keyword%20&sl=auto&tl=hu&dj=1&source=bubble"

so, basically it says that they suspect unusual activity from my ip or network that may violate the TOS.

then I checked on the extension page and seems that i'm not the only one, and finally I got here

hope it helps

cheers

**Comment by artemave** [comment]
Thanks @atrox3d , this must be it. TransOver totally violates their TOS so fair enough. I think being on VPN might fix it for you.

**Comment by artemave** [comment]
Phone tethering might also do the trick :)

**Comment by rziman** [comment]
I've also experienced the issue described so far in this thread (as well as at the support tab at the chrome web store: https://chrome.google.com/webstore/detail/transover/aggiiclaiamajehmlfpkjmlbadmkledi/support?hl=en). Below are a few additional notes and an in-place alternative for anyone that needs an immediate workaround with similar functionality without setting up a VPN.

The issue seems to affect Google's own official Google Translate plugin (https://chrome.google.com/webstore/detail/google-translate/aapbdbdomjkkjkaonfhkkikfgjllcleb/). In the following screenshot, the TransOver popup bubble appears next to the Google Translate plugin popup:

![screenshot at 2018-08-01 14 07 51](https://user-images.githubusercontent.com/868248/43523358-703a0af2-959c-11e8-9044-abaafac2a61b.png)

After a minute or so of frequent usage, neither plugin works anymore, and the console log for the TransOver background page shows the "unusual traffic from your computer network" response from translate.googleapis.com. The console log for the Google Translate plugin doesn't seem to be as detailed, but a quick grep through its source code shows that it's using the same API. (The "gtx" in the "client=gtx" part of the query string which appears in the Google Translate source - and also in the current TransOver source - probably originally stood for "Google Translate eXtension".) It seems to take at least 15-20 minutes for the API to become responsive again, though it might have some sort of backoff function so that it takes longer and longer each time. Either way, the rate limiter makes the plugins completely useless while it's on.

Maybe the Google Translate team changed some parameters in the rate limiter recently, but the issue isn't altogether completely new. See the following Oct 2015 thread reporting the same issue for a plugin that made use of the text-to-speech API:

Script not working due to Google rate limitation ("unusual traffic" message) · Issue #4 · glutanimate/simple-google-tts
https://github.com/Glutanimate/simple-google-tts/issues/4

The Google product forums also show both recent (Jul 2018) and old (Oct 2015) posts reporting the same issue.

unusual traffic (in Google Translate Forum) - Google Product Forums
https://productforums.google.com/forum/#!topicsearchin/translate/unusual$20traffic

Now for the workarounds. Note that with the official Google Translate plugin, you can still right-click and have an option to feed the highlighted text to translate.google.com in a separate tab (of course, the whole point of TransOver was to have a HUD-like popup that eliminates this cumbersome step):

![screenshot at 2018-08-01 14 11 42](https://user-images.githubusercontent.com/868248/43524402-85565474-959f-11e8-92f1-a83364039fcd.png)

For a full workaround that preserves the popup functionality, an in-place alternative here is to use the Google Dictionary plugin (https://chrome.google.com/webstore/detail/google-dictionary-by-goog/mgijmajocgfcbeboacabfgobmjgjcoja/related). It appears to do translation as well if you set it to a language different than the one highlighted:

![screenshot at 2018-08-01 14 48 23](https://user-images.githubusercontent.com/868248/43524607-fd09beac-959f-11e8-820a-f9978dfe21ca.png)

It also has a downloadable word history feature that preserves the translations. The caveat is that it doesn't format the popup cleanly when more than few words are selected:

![screenshot at 2018-08-01 14 49 32](https://user-images.githubusercontent.com/868248/43524631-0da8c8f2-95a0-11e8-9708-0464da813579.png)

I've only been using this Google Dictionary plugin for a couple hours, so I'm not sure if it doesn't have its own rate limiting issue. There are some additional options (search "alternatives to google translate"), but I haven't tried them.

PS @artemave: I found TransOver to be superb and very useful before hitting up against this rate limiter issue. I'd be happy to help find an implementation workaround to properly fix it and then help support it, but I'm not familiar with JQuery and chrome plugins generally and could use a bit of guidance on the development workflow (through a short Skype or Hangouts call, say). If this is something you're open to, let me know.

**Comment by mozkan007** [comment]
Hi I have the same problem. Transover can not translate because of the Google's unusual traffic problem. I hope you can fix it soon because I always use transover and without it I can not use any English websites. Thanks.

**Comment by artemave** [comment]
I published a new version - 1.42 - and I believe the issue is resolved. At least for the time being.

Please give it a test and report back if you're still seeing it.

Last but not least, thank you very much for identifying the problem.

**Comment by atrox3d** [comment]
Hello, I gave it a quick test and it seems to work again. Wonderful job!
I'll keep you posted.
Thank you again for your effort.

![2018-08-05_203432](https://user-images.githubusercontent.com/13483498/43688745-1229aade-98ef-11e8-9702-b7e5f495f278.png)
